File: Medium/Java/Coin_Change_2(518).java
public class Solution {
    public int change(int amount, int[] coins) {
        int[][] results = new int[coins.length + 1][amount + 1];
        
        results[0][0] = 1;
      
        for (int i = 1; i <= coins.length; i++) {
            results[i][0] = 1;
            
            for (int j = 1; j <= amount; j++) {
                results[i][j] = results[i-1][j] + (j >= coins[i-1] ? results[i][j - coins[i-1]] : 0);
            }
        }
        
        return results[coins.length][amount];
    }
}

File: Medium/C++/Coin_Change_2(518).cpp
class Solution {
public:
	int change(int amount, vector<int>& c) {
		ios::sync_with_stdio(false);
		vector<int> dp(amount+1);
		dp[0] = 1;
		for(auto a : c){
			for(int i = a ; i <= amount ; ++i)
				dp[i] += dp[i-a];
		}
		return dp[amount];
    }
};